engine/src/main/java/org/entando/entando/web/database/DatabaseController.java

```diff
@@ -92,6 +92,17 @@
         return new ResponseEntity<>(new RestResponse(response), HttpStatus.OK);
     }
 
+    @RestAccessControl(permission = Permission.SUPERUSER)
+    @RequestMapping(value = "/restoreBackup/{reportCode}", method = RequestMethod.PUT, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<RestResponse> restoreBackup(@PathVariable String reportCode) throws Throwable {
+        logger.debug("Starting database restore");
+        this.getDatabaseService().startDatabaseRestore(reportCode);
+        Map<String, String> response = new HashMap<>();
+        response.put("status", String.valueOf(this.getDatabaseService().getStatus()));
+        logger.debug("Database restore started");
```

Comment:
This should probably say completed. We should log the report code

Reply:
Done!